Automattic/woocommerce-payments
# feat: tokenized cart PRBs on PDPs

Fixes https://github.com/Automattic/woocommerce-payments/issues/8526

#### Changes proposed in this Pull Request









Implementing the tokenized cart on PDPs, as a first step.
This is behind the `_wcpay_feature_tokenized_cart_prb` feature flag. Enable it in your `wp_options`.

There are a few `TODO`s that have been inherited from the previous implementation or that will be useful for the future implementations.

#### Testing instructions





* Enable the "Enable Cart-Token implementation for PRBs" flag in your dev tools (you'll need to update them)
* Ensure you have Google Pay/Apple Pay enabled in the merchant's settings
* Go to a product page on your site
* You should be able to use the PRB button to get shipping data & check out on simple products & variable products
* There shouldn't be a difference in functionality with the older implementation (when the "Enable Cart-Token implementation for PRBs" functionality is disabled) besides what I wrote below - if you do notice something weird, please let me know!
  * The implementation allows you to check out with the US shipping address, but for some reason the UK and CA addresses aren't working correctly (they fail validation in the backend, despite the values being correct)
  * The itemized amount (tax, shipping, etc.) is _sometimes_ not displayed
  * The selected currency is being ignored, and the store's default currency is used instead ( Fixed by https://github.com/Automattic/woocommerce-payments/pull/8816 )
  * The payment method being reported  in the "Order success" and "Order details" pages is inaccurate (Fixed by https://github.com/Automattic/woocommerce-payments/pull/8842 )
<img width="292" alt="Screenshot 2024-05-02 at 6 26 45 PM" src="https://github.com/Automattic/woocommerce-payments/assets/273592/437dd4c1-69e0-4500-8b55-32514f94eb99">

https://github.com/Automattic/woocommerce-payments/assets/273592/ce236367-d2e1-4e01-bd34-d5c4c77442c7



-------------------


**Post merge**
